ocaml/xapi/xapi_globs.ml

```diff
@@ -951,7 +951,10 @@ let ignore_vtpm_unimplemented = ref false
 
 let evacuation_batch_size = ref 10
 
-type xapi_globs_spec_ty = Float of float ref | Int of int ref
+type xapi_globs_spec_ty =
+  | Float of float ref
+  | Int of int ref
+  | TimeSpanOfFloat of Mtime.Span.t ref
```

Comment:
```suggestion
  | TimeSpanOfSeconds of Mtime.Span.t ref
```

Reply:
The purpose here is for `TimeSpanOfInt` in the future change, as I notice that there are lots of timers/intervals are read from `int` as well.
Is it possible to solve this with typing?

Reply:
How about:
```
type time_span =
  | OfFloat of Mtime.Span.t ref
  | OfInt of Mtime.Span.t ref

type xapi_globs_spec_ty =
  | Float of float ref
  | Int of int ref
  | TimeSpanOfSeconds of time_span

...

; ("host_assumed_dead_interval", TimeSpanOfSeconds (OfFloat host_assumed_dead_interval))

...

match ty with
    | Float x ->
        Arg.Set_float x
    | Int x ->
        Arg.Set_int x
    | TimeSpanOfSeconds (OfFloat x) ->
        Arg.Float
          (fun y ->
            let y' = Float.to_int (1000. *. y) in
            x := Mtime.Span.(y' * ms)
          )
    | TimeSpanOfSeconds (OfInt x) ->
        ....
```

Reply:
> I notice that there are lots of timers/intervals are read from int as well.

Not sure which ones you mean, they usually can be expressed with floats as well. One good reason to use integers would be because conversion from float has a limit on precision, and more than 104 days cannot be expressed.

At that point I would be inclined to express the two uses by types:

```ocaml
type 'a xapi_globs_spec =
  | Float of float ref
  | Int of int ref
  | ShortDurationInSeconds of Mtime.Span.t ref
  | LongDurationInSeconds of Mtime.Span.t ref

...

      , ( match ty with
        | Float x ->
            Arg.Set_float x
        | Int x ->
            Arg.Set_int x
        | ShortDurationInSeconds x ->
           Arg.Float (fun y -> x := Clock.Timer.s_to_span y)
        | LongDurationInSeconds x ->
            Arg.Int (fun y -> x := Mtime.Span.(y * s))
        )
```